### PR TITLE
**Fix:** Handle outside links in 'to' props

### PR DESCRIPTION
**Pull request by peterszerzo**
## Summary

Currently, `<Button to="https://github.com/something" />` still attempts to do a pushstate. This PR makes it so that the default action is not prevented if an outside link is recognized.

## To be tested

Me
- [x] No error/warning in the console on `localhost:6060`
- [x] `pushState` links work
- [x] `pushState` isn't used if an outside url is added (Button, Breadcrumb, SidenavHeader, SidenavItem, Logo)

Tester 1 (Imo)

- [x] The netlify build is working
- [x] `pushState` links work
- [x] `pushState` isn't used if an outside url is added (Button, Breadcrumb, SidenavHeader, SidenavItem, Logo)

Tester 2

- [x] The netlify build is working
- [x] `pushState` links work
- [x] `pushState` isn't used if an outside url is added (Button, Breadcrumb, SidenavHeader, SidenavItem, Logo)

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ export { default as wrapDefaultTheme } from "./wrap-default-theme"
 
 export const isModifiedEvent = (event: any) => !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey)
 
+export const isOutsideLink = (url: string) => Boolean(url.match(/^http/g))
```
Comment (TejasQ) [inline src/utils/index.ts]: Let's lean on the shoulders of open-source giants for a truly tested solution for this. https://www.npmjs.com/package/url-regex, for example, in order to best handle as many of the possible edge cases as we can.

**Comment by TejasQ** [comment]
Is this a `**Fix**:`?

**Comment by peterszerzo** [comment]
@TejasQ yes it is - I've just updated the title.

File: src/utils/index.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
+import urlRegex from "url-regex"
+
 export { default as deprecate } from "./deprecate"
 export * from "./mixins"
 export * from "./color"
 export { default as wrapDefaultTheme } from "./wrap-default-theme"
 
 export const isModifiedEvent = (event: any) => !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey)
 
+export const isOutsideLink = (url: string) => Boolean(urlRegex.test(url, { exact: true }))
```
Comment (TejasQ) [inline src/utils/index.ts]: It seems `urlRegex.test` isn't a property based on the checks.

Comment (peterszerzo) [reply src/utils/index.ts]: Yes, it should have been `urlRegex()` - I just updated

**Pull request merged**